x/merkledb/proof.go

```diff
@@ -671,15 +669,14 @@ func verifyKeyChanges(kvs []KeyChange, start, end []byte) error {
 // If [start] is nil, there is no lower bound on acceptable keys.
 // If [end] is nil, there is no upper bound on acceptable keys.
 // If [kvs] is empty, returns nil.
-func verifyKeyValues(kvs []KeyValue, start, end []byte) error {
+func verifyKeyValues(kvs []KeyValue, start []byte, end Maybe[[]byte]) error {
```

Comment:
could we update the comment for this fn to refer to Nothing instead of nil as appropriate?